dkc/core/views.py

```diff
@@ -1,23 +1,20 @@
 from django.contrib.admin.views.decorators import staff_member_required
-from django.db import connection
+from django.db.models import Count, F
 from django.http import HttpRequest, HttpResponse
 from django.shortcuts import render
 
-TREES_BY_FILE_COUNT_Q = """
-SELECT folder.id, folder.name, sq.c as file_count, pg_size_pretty(folder.size) as size
-FROM core_folder as folder, (
-  SELECT tree.id as tree_id, count(*) as c
-  FROM core_file as file, core_folder as folder, core_tree as tree
-  WHERE file.folder_id = folder.id
-  AND folder.tree_id = tree.id
-  GROUP BY tree.id
-) as sq WHERE folder.tree_id = sq.tree_id AND folder.parent_id IS NULL ORDER BY sq.c DESC
-"""
+from dkc.core.models import Tree
 
 
 @staff_member_required
 def trees_by_file_count_view(request: HttpRequest) -> HttpResponse:
-    with connection.cursor() as cursor:
-        cursor.execute(TREES_BY_FILE_COUNT_Q)
-        rows = cursor.fetchall()
-    return render(request, 'staff/trees_by_file_count.html', {'rows': rows})
+    trees_annotated = (
+        Tree.objects.annotate(
+            size=F('quota__used'),
```

Comment:
`quota__used` is not the correct thing to use for the size of the tree, as multiple trees can share the same quota.

Reply:
Fixed.